resources/api/reservation.py

```diff
@@ -167,7 +193,7 @@ def validate(self, data):
         if not resource.can_make_reservations(request_user):
             raise PermissionDenied(_('You are not allowed to make reservations in this resource.'))
 
-        if data['end'] < timezone.now():
+        if data.get('end', False) and data['end'] < timezone.now():
```

Comment:
`if 'end' in data` is more explicit for testing key existence than `data.get()`. Mostly matters for readability. I had to think for a bit why `.get` is used before I understood this line.

Reply:
Agreed, fix'd